src/components/HomeUpload/HomeUpload.js

```diff
@@ -11,8 +11,33 @@ import AppContext from "../../AppContext";
 export default function HomeUpload() {
   const [files, setFiles] = useState([]);
   const { apiUrl } = useContext(AppContext);
+  const [directoryMode, setDirectoryMode] = useState(false);
+
+  useEffect(() => {
+    if (directoryMode) {
+      inputRef.current.setAttribute("webkitdirectory", "true");
+    } else {
+      inputRef.current.removeAttribute("webkitdirectory");
+    }
+  }, [directoryMode]);
+
+  const getFilePath = (file) => {
+    const filePath = file.path || file.webkitRelativePath || file.name;
```

Comment:
Is it safer to do `file.webkitRelativePath || file.path || file.name;`?

Reply:
```suggestion
    const filePath = file.webkitRelativePath || file.path || file.name;
```